platform/darwin/src/MGLFeature.h

```diff
@@ -0,0 +1,133 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLPolyline.h"
+#import "MGLPolygon.h"
+#import "MGLPointAnnotation.h"
+#import "MGLShapeCollection.h"
+
+#import "MGLTypes.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ The `MGLFeature` protocol is used to provide details about geographic features
+ contained in a map view’s vector tile sources. Each concrete subclass of
+ `MGLShape` in turn has a subclass that conforms to this protocol.
+ 
+ Typically, you do not create feature objects yourself but rather obtain them
+ using `-[MGLMapView visibleFeaturesAtPoint:]` and related methods. Each feature
+ object associates a shape with an identifier and attributes as specified by the
+ source. Like ordinary `MGLAnnotation` objects, some kinds of `MGLFeature`
+ objects can also be added to a map view using `-[MGLMapView addAnnotations:]`
+ and related methods.
+ */
+@protocol MGLFeature <MGLAnnotation>
+
+/**
+ An object that uniquely identifies the feature in its vector tile source.
+ 
+ The value of this property is currently always an `NSNumber` object but may in
+ the future be an instance of another class, such as `NSString`.
+ 
+ The identifier corresponds to the
+ <a href="https://github.com/mapbox/vector-tile-spec/tree/master/2.1#42-features">feature identifier</a>
+ (`id`) in the vector tile source. If the source does not specify the feature’s
+ identifier, the value of this property is `nil`.
+ 
+ For details about the identifiers used in most Mapbox-provided styles, consult
+ the
+ <a href="https://www.mapbox.com/vector-tiles/mapbox-streets/">Mapbox Streets</a>
+ layer reference.
+ */
+@property (nonatomic, copy, nullable, readonly) id identifier;
+
+/**
+ A dictionary of attributes specified by the vector tile source for this
+ feature.
+ 
+ The keys and values of this dictionary are determined by the vector tile
+ source. Vector tiles are stored in <a href="http://geojson.org/">GeoJSON</a>
+ format; they additionally conform to the
```

Comment:
Oops, that’s incorrect and it’s the next part, about the vector tile spec, that we should be emphasizing anyways.